util/attendance/src/lib.rs

```diff
@@ -0,0 +1,65 @@
+#![no_std]
+
+#[allow(unused_imports)]
+use multiversx_sc::imports::*;
+use multiversx_sc::{
+    codec,
+    proxy_imports::{NestedDecode, NestedEncode},
+};
+use multiversx_sc::{
+    derive::type_abi,
+    proxy_imports::{TopDecode, TopEncode},
+};
+
+#[type_abi]
+#[derive(TopEncode, TopDecode, NestedDecode, NestedEncode)]
+pub struct StudentStruct<M: ManagedTypeApi> {
+    pub address: ManagedAddress<M>,
+    pub name: ManagedBuffer<M>,
+}
+
+#[multiversx_sc::contract]
+pub trait Attendance {
+    #[init]
+    fn init(&self) {}
+
+    #[only_owner]
+    #[upgrade]
+    fn upgrade(&self) {}
+
+    #[endpoint(registerStudent)]
+    fn register_student(&self) {
+        let caller = self.blockchain().get_caller();
+
+        require!(
+            self.students().contains(&caller),
+            "Student is already registered"
+        );
+
+        self.students().insert(caller);
+    }
+
+    #[endpoint(registerAttendance)]
+    fn register_attendance(&self) {
```

Comment:
What if a students call `register_attendance` 13 times in the last week?
I would add a `last_attendance_epoch` storage and make sure that between this call's epoch and the `last_attendance_epoch` is at least 6.
Or maybe blocks are better to be used because epoch changes a bit.